x-pack/plugins/stack_connectors/server/connector_types/auth/schema.ts

```diff
@@ -55,7 +55,13 @@ export const SecretConfigurationSchemaValidation = {
   },
 };
 
-export const SecretConfigurationSchema = schema.object(
-  SecretConfiguration,
+export const SecretConfigurationSchema = SecretConfiguration.extends(
+  {},
   SecretConfigurationSchemaValidation
 );
+
+export type HasAuth = TypeOf<typeof hasAuthSchema>;
```

Comment:
Could you please preserve the initial file structure and keep these in a separate `auth/types.ts`? 🙏

Reply:
ACK! I'm not sure if that'll defeat the purpose since the types are based on a _runtime_ object. I'll play around with this to see if there's a more elegant way :)